File: pset6/sentimental-readability/readability.py
from cs50 import get_string
import sys


def main():
    text = get_string("Text: ")

    lettersNum = countLetters(text)
    sentencesNum = countSentences(text)
    wordsNum = countWords(text)
    l = (lettersNum / wordsNum) * 100
    s = (sentencesNum / wordsNum) * 100
    grade = colemanIndex(l, s)

    if grade > 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print("Grade " + str(grade))


def countLetters(text):
    counter = 0

    for letter in text:
        if letter.isalpha():
            counter += 1

    return counter


def countWords(text):
    counter = 1
    for letter in text:
        if(letter == " "):
            counter += 1

    return counter


def countSentences(text):
    counter = 0
    for letter in text:
        if(letter == '.' or letter == '?' or letter == '!'):
            counter += 1
    return counter


def colemanIndex(l, s):
    return round(0.0588 * l - 0.296 * s - 15.8)


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);
int colemanIndex(float L, float S);

int main(void)
{
    string text = get_string("Text: ");

    int letters = countLetters(text);
    int sentences = countSentences(text);
    int words = countWords(text);

    // index = 0.0588 * L - 0.296 * S - 15.8
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    int grade = colemanIndex(L, S);

    //printf("%d, %d, %d, %f, %f", letters, sentences, words, L, S);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

int countLetters(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            counter++;
        }
    }
    return counter;
}

int countWords(string text)
{
    int counter = 0;
    char *pointer = strtok(text, " ");

    while (pointer != NULL)
    {
        pointer = strtok(NULL, " ");
        counter++;
    }
    return counter;
}

int countSentences(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            counter++;
        }
    }
    return counter;
}

int colemanIndex(float L, float S)
{
    return (int) round(0.0588 * L - 0.296 * S - 15.8);
}